src/stream.jl

```
@@ -489,21 +489,21 @@ function Base.write(stream::TranscodingStream)
     return 0
 end
 
-function Base.write(stream::TranscodingStream, b::UInt8)
+function Base.write(stream::TranscodingStream, b::UInt8)::Int
     changemode!(stream, :write)
-    if marginsize(stream.buffer1) == 0 && flushbuffer(stream) == 0
-        return 0
```

`write` will throw an error or block instead of returning 0 on failure.